skippkg-finder.py

```diff
@@ -0,0 +1,333 @@
+        """
+        Returns binarylist of a project
+        """
+
+        if project.startswith('SUSE:'):
+            repository = 'pool'
+
+        path = ['build', project, repository, arch]
+        url = makeurl(self.apiurl, path, {'view': 'binaryversions'})
+        root = ET.parse(http_GET(url)).getroot()
+
+        for binary_list in root:
+            package = binary_list.get('package')
+            package = package.split(':', 1)[0]
+            index = project + "_" + package
+
+            if index not in package_binaries:
+                package_binaries[index] = []
+            for binary in binary_list:
+                filename = binary.get('name')
+                result = re.match(RPM_REGEX, filename)
+                if not result:
+                    continue
+
+                if result.group('arch') == 'src' or result.group('arch') == 'nosrc':
+                    continue
+                if result.group('name').endswith('-debuginfo') or result.group('name').endswith('-debuginfo-32bit'):
+                    continue
+                if result.group('name').endswith('-debugsource'):
+                    continue
+
+                if result.group('name') not in package_binaries[index]:
+                    package_binaries[index].append(result.group('name'))
+
+        return package_binaries
+
+    def item_exists(self, project, package=None):
+        """
+        Return true if the given project or package exists
+        """
+
+        if package:
+            url = makeurl(self.apiurl, ['source', project, package, '_meta'])
+        else:
+            url = makeurl(self.apiurl, ['source', project, '_meta'])
+        try:
+            http_GET(url)
+        except HTTPError:
+            return False
+        return True
+
+    def origin_metadata_get(self, project, package):
+        """
+        Returns origin infomration
+        """
+
+        meta = ET.fromstringlist(osc.core.show_package_meta(self.apiurl, project, package))
+        if meta is not None:
+            return meta.get('project'), meta.get('name')
+
+        return None, None
+
+    def get_linkinfo(self, project, package):
+        """
+        Returns package links if it does exist
+        """
+
+        query = {'withlinked': 1}
+        u = makeurl(self.apiurl, ['source', project, package], query=query)
+        root = ET.parse(http_GET(u)).getroot()
+        linkinfo = root.find('linkinfo')
+        if linkinfo is not None:
+            return linkinfo.get('package')
+
+        return package
+
+    def exceptions(self, package):
+        """
+        Do not skip the package if matches the condition
+        """
+
+        if package.startswith('python2') or package.startswith('python3') or \
+                package.startswith('preinstallimage-'):
+            return True
+        if package.endswith('-bootstrap'):
+            return True
+        if 'Tumbleweed' in package or 'metis' in package:
+            return True
```

Comment:
I expect there rationale for non obvious exceptions like metis or python

Reply:
there are some python pacakge wasn't single-spec'ed, eg. python-abc is python3 only, and there is python2-abc for the compatibility reason. metis is a special case since it was conflicting to scotch however we can not really delete a package in SLE, see https://bugzilla.suse.com/show_bug.cgi?id=1182799

Reply:
thanks for explanation. It would be nice to have in code so we do not forget in future when the need to touch script appear.